src/Bicep.Core/Analyzers/Linter/Rules/SecretsInParamsMustBeSecureRule.cs

```diff
@@ -62,20 +64,24 @@ override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
         private IDiagnostic? AnalyzeUnsecuredParameter(ParameterSymbol parameterSymbol)
         {
             string name = parameterSymbol.Name;
-            if (HasSecretRegex.IsMatch(name))
+            string type = parameterSymbol.Type.ToString();
+            if (IsType.IsMatch(type))
```

Comment:
@majastrz @anthony-c-martin What's the correct way to check if a parameter is of type object or string?  Is simply checking the TypeKind enough (maybe for Object but not String?)?  Or is TypeValidator.AreTypesAssignable needed?

I do have this helper:
```cs
namespace Bicep.Core.Analyzers.Linter.Common
{
    public static class TypeExtensions
    {
        /// <summary>
        /// True if the given type symbol is a string type (and not "any")
        /// </summary>
        public static bool IsStrictlyAssignableToString(this TypeSymbol typeSymbol)
        {
            return typeSymbol is not AnyType
                && TypeValidator.AreTypesAssignable(typeSymbol, LanguageConstants.String);
        }
    }
}
```
But frankly I'm not positive when Any would come into play.  Thanks!

Reply:
`TypeValidator.AreTypesAssignable` feels like the simplest option. Technically a parameter type cannot be `any` (since the type is always declared)

Reply:
Fixed in https://github.com/dciborow/bicep/pull/2